pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageCrypto.java

```diff
@@ -487,14 +505,14 @@ private ByteBuf decryptData(SecretKey dataKeySecret, MessageMetadata msgMetadata
         return targetBuf;
     }
 
-    private ByteBuf decryptDataKeyAndData(MessageMetadata msgMetadata, ByteBuf payload, CryptoKeyReader keyReader) {
+    private ByteBuf getKeyAndDecryptData(MessageMetadata msgMetadata, ByteBuf payload) {
 
         ByteBuf decryptedData = null;
 
-        List<KeyByteValue> encKeys = msgMetadata.getEncryptionKeysList();
+        List<EncryptionKeys> encKeys = msgMetadata.getEncryptionKeysList();
 
         // Go through all keys to retrieve data key from cache
-        for(int i=0; i<encKeys.size(); i++) {
+        for (int i = 0; i < encKeys.size(); i++) {
```

Comment:
`for(KeyByteValue encKey: encKeys) {..} ..?`

Reply:
It needs to set decryptedData which can't be done from the lambda